src/backend/src/controllers/projects.controllers.ts

```diff
@@ -1,341 +1,101 @@
-    changes.push(bomChangeJson);
-  }
-  if (taskChangeJson !== undefined) {
-    changes.push(taskChangeJson);
-  }
-  if (projectManagerChangeJson !== undefined) {
-    changes.push(projectManagerChangeJson);
-  }
-  if (projectLeadChangeJson !== undefined) {
-    changes.push(projectLeadChangeJson);
-  }
-
-  // Dealing with lists
-  const rulesChangeJson = createRulesChangesJson('rules', originalProject.rules, rules, crId, userId, wbsElementId);
-  const goalsChangeJson = createDescriptionBulletChangesJson(
-    originalProject.goals.filter((element) => !element.dateDeleted).map((element) => descBulletConverter(element)),
-    goals,
-    crId,
-    userId,
-    wbsElementId,
-    'goals'
-  );
-  const featuresChangeJson = createDescriptionBulletChangesJson(
-    originalProject.features.filter((element) => !element.dateDeleted).map((element) => descBulletConverter(element)),
-    features,
-    crId,
-    userId,
-    wbsElementId,
-    'features'
-  );
-  const otherConstraintsChangeJson = createDescriptionBulletChangesJson(
-    originalProject.otherConstraints
-      .filter((element) => !element.dateDeleted)
-      .map((element) => descBulletConverter(element)),
-    otherConstraints,
-    crId,
-    userId,
-    wbsElementId,
-    'other constraints'
-  );
-  // add the changes for each of dependencies, expected activities, and deliverables
-  changes = changes
-    .concat(rulesChangeJson)
-    .concat(goalsChangeJson.changes)
-    .concat(featuresChangeJson.changes)
-    .concat(otherConstraintsChangeJson.changes);
 
-  // update the project with the input fields
-  const updatedProject = await prisma.project.update({
-    where: {
-      wbsElementId
-    },
-    data: {
-      budget,
-      summary,
-      googleDriveFolderLink,
-      slideDeckLink,
-      bomLink,
-      taskListLink,
-      rules,
-      wbsElement: {
-        update: {
-          name,
-          projectLeadId: projectLead,
-          projectManagerId: projectManager
-        }
-      }
+  static async editProject(req: Request, res: Response, next: NextFunction) {
+    try {
+      const user = await getCurrentUser(res);
+      const {
+        projectId,
+        crId,
+        name,
+        budget,
+        summary,
+        rules,
+        goals,
+        features,
+        otherConstraints,
+        googleDriveFolderLink,
+        slideDeckLink,
+        bomLink,
+        taskListLink,
+        projectLead,
+        projectManager
+      } = req.body;
+
+      const editedProject: Project = await ProjectsService.editProject(
+        user,
+        projectId,
+        crId,
+        name,
+        budget,
+        summary,
+        rules,
+        goals,
+        features,
+        otherConstraints,
+        googleDriveFolderLink,
+        slideDeckLink,
+        bomLink,
+        taskListLink,
+        projectLead || null,
+        projectManager || null
+      );
+
+      return res.status(200).json(editedProject);
```

Comment:
In the editWorkPackage function, we return a string. Maybe we should do that here too?

<img width="593" alt="Screen Shot 2023-01-24 at 10 31 29 PM" src="https://user-images.githubusercontent.com/60333735/214474038-6f84c123-8f72-4258-95e9-4f3c57952c4b.png">

Reply:
tbh the string responses are worthless so this can only be helpful

(it's possible the frontend needs the updated project or i just added it for fun and forgot but either way it can't be bad)